components/render/sdl2backend.cpp

```diff
@@ -818,203 +820,117 @@ namespace Render
         drawMinPillar(s, x, y, pillar, tileset, false);
     }
 
-    void drawAt(const Level::Level& level, const Sprite& sprite, int32_t x1, int32_t y1, int32_t x2, int32_t y2, size_t dist, int32_t levelX, int32_t levelY)
-    {
-        size_t w, h;
-        spriteSize(sprite, w, h);
-
-        int32_t xPx1 = ((y1*(-32)) + 32*x1 + level.width()*32) + levelX - w/2;
-        int32_t yPx1 = ((y1*16) + (16*x1) +160) + levelY;
-
-        int32_t xPx2 = ((y2*(-32)) + 32*x2 + level.width()*32) + levelX - w/2;
-        int32_t yPx2 = ((y2*16) + (16*x2) +160) + levelY;
+    // basic transform of isometric grid to normal, (0, 0) tile coordinate maps to (0, 0) pixel coordinates
+    // since eventually we're gonna shift coordinates to viewport center, it's better to keep transform itself
+    // as simple as possible
+    static Point tileTopPoint (const Tile &tile) {
+      return {(tileWidth / 2) * (tile.x - tile.y), (tile.y + tile.x) * (tileHeight / 2)};
+    }
 
-        int32_t x = xPx1 + ((((float)(xPx2-xPx1))/100.0)*(float)dist);
-        int32_t y = yPx1 + ((((float)(yPx2-yPx1))/100.0)*(float)dist);
+    // this function simply does the reverse of the above function, could be found by solving linear equation system
+    // it obviously uses the fact that ttileWidth = tileHeight * 2
+    Tile getTileFromScreenCoords(const Point& screenPos, const Vector& toScreen)
```

Comment:
I do believe it technically does the same thing but likely +/- some constant (x,y) which is irrelevant because we center everything on a player in the end anyway. I don't know if it was 'overcomplicated' but it contained at least some ifs and this version is pure arithmetics with division by 64 which is obviously optimized to >> 6. So it should perform nicer.